token-minter/tokenminter.go

```diff
@@ -35,7 +35,20 @@ const ErrorRetryPeriod = 10 * time.Second
 
 func NewStartCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "token-minter",
+		Use:   "token-minter",
+		Short: "Mint ServiceAccount tokens to the filesystem for ServiceAccounts in a tenant cluster.",
+		Long: `Components running on the service cluster with connectivity to tenant clusters need to run
```

Comment:
@enxebre I added this help doc because there's otherwise no doc I could find for what all these components do, and it took me a while to ramp up due to that. Given what you've written about `token-minter`, I know the doc I wrote is incomplete at best. Please help me make this doc better :)

Reply:
huge thanks for taking care of documenting this. What you have put together sgtm, I'm leaving here some additional context in case you want to complement it.

The token minter is a binary that receives a kubeconfig and implement the following operations against it :
- Creates a SA if it doesn't exist.
- Uses the [tokeRequest API](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/) to create a token for that SA.
- Write the token to disk.

Now, how we leverage the above workflow:
**Use case 1:**
Every HCP KAS is configured with a [ServiceAccountIssuer](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) that points to a s3 object URL that satisfies [the OpenID Discovery spec](https://github.com/openshift/hypershift/blob/bb04b0a79621406a3c0b7522e50ee51cce70f85f/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L3929-L3956) and which is also backing a pre-created [AWS identityProvider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers.html) for that HCP.

Because of the above setup, SA minted tokens can be [used in a AWS creds config files](https://github.com/openshift/hypershift/blob/bb04b0a79621406a3c0b7522e50ee51cce70f85f/hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go#L242-L244) to authenticate against AWS STS and [assume roles temporarily with AssumeRoleWithWebIdentity](https://github.com/openshift/hypershift/blob/bb04b0a79621406a3c0b7522e50ee51cce70f85f/cmd/infra/aws/iam.go#L745-L762)
https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html

**Use case 2:**
The SA minted token can be used to create a client against the kubeconfig cluster that would happen to be the tenant cluster fort HCP.
Afaik this use case is only used by registry operator atm and collides with ReconcileServiceAccount, so unless there's a reason to keep both I'd consolidate on the former.

cc @sjenning which originally wrote the token minter I believe.